beancount_import/source/generic_importer_source.py

```diff
@@ -84,24 +85,20 @@ def prepare(self, journal: 'JournalEditor', results: SourceResults) -> None:
             make_import_result=self._make_import_result,
             results=results)
 
-    def _add_description(self, entry: Transaction):
+    def _add_description_and_unconfirmed_account_key(self, entry: Transaction):
         if not isinstance(entry, Transaction): return None
-        postings = entry.postings #type: List[Posting]
-        to_mutate = []
-        for i, posting in enumerate(postings):
-            if posting.account != self.account: continue
-            if isinstance(posting.meta, dict):
+        for i, posting in enumerate(entry.postings):
+            if not isinstance(posting.meta, dict):
+                # replace posting with a clone with meta dict
+                entry.postings[i] = posting._replace(meta={})
+                posting = entry.postings[i]
+            if posting.account == self.account:
+                # add description
                 posting.meta["source_desc"] = entry.narration
                 posting.meta["date"] = entry.date
-                break
             else:
-                to_mutate.append(i)
-                break
-        for i in to_mutate:
-            p = postings.pop(i)
-            p = Posting(p.account, p.units, p.cost, p.price, p.flag,
-                        {"source_desc":entry.narration, "date": entry.date})
-            postings.insert(i, p)
+                # add unconfirmed account key
+                posting.meta[UNCONFIRMED] = ""
```

Comment:
It looks like currently all accounts other than `self.account` are considered to be editable.  Should there be a way for the importer to indicate a non-editable account?

Reply:
Wouldn't it be hard to differentiate without smart_importer adding some meta to the postings? I can check for `"confirmed" : True` key in a posting meta before adding the unconfirmed key. So if an importer or `smart_importer` adds 2nd/nth posting abd wants to make a posting non-editable, they can add the confirmed tag, but there wouldn't be a way to enforce it.

Reply:
I personally don't yet see the utility of this. Even if a posting is marked as editable by the generic_importer, one can safely ignore it in the beancount-import interface. I admit it is slightly more cognitively taxing to have to review an extra candidate posting.

Reply:
Yes, the idea would be to reduce cognitive load.  Accounts that are known from the structure of the data itself (e.g. investment transactions where some legs are known to be interest, dividends, capital gains, fees, etc.) don't need to be inspected manually, while any "guessed" accounts may well be wrong and need to be manually checked somewhat carefully, which takes longer and more mental effort.

I suppose currently smart_importer does not provide any indication of which accounts were "guessed" vs determined from the structure of the data?  Could it be made to do so, i.e. add unconfirmed_account: True?  Then it would not be necessary for beancount-import to add it.

It also seems like it would be better if the unconfirmed_account metadata field were stripped out before actually writing the transaction to the journal, and it would be kind of confusing to leave it after the account has actually been confirmed.  The exception would be the FIXME action --- that should leave unconfirmed_account in place.  Then the presence of unconfirmed_account in the journal should be treated like an Expenses:FIXME account, in that it should be presented to the user as a pending transaction.

In general it might be easier if one of you, @Zburatorul or @dumbPy, could combine this pull request with #60 as a single pull request (and ideally add an example and test of using this with smart_importer if that is the ultimate goal), so that it is easier to test that it all actually works together in a desirable way.

Reply:
I see the logic now. Shall we for now move forward assuming that smart_importer will insert the unconfirmed_account key for any uncertain guesses? The absence will be assumed to mean certainty.

I implemented the logic you suggest of removing the "unconfirmed_account" key from postings with accounts other than FIXME upon acceptance in the UI.

@dumbPy, since my changes are smaller and simpler than yours I suggest you cherry-pick them onto your branch and modify at will.